hawkular-actions-plugins/hawkular-actions-email/src/main/java/org/hawkular/actions/email/template/EmailTemplate.java

```diff
@@ -0,0 +1,259 @@
+            if (templateHawkularUrl == null) {
+                templateHawkularUrl = System.getenv(EmailPlugin.HAWKULAR_BASE_URL);
+            }
+            templateData.put("url", templateHawkularUrl);
+
+            String templateLocale = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_LOCALE);
+            Condition.Type type = getFirstConditionType(alert);
+            if (type != null) {
+                templateData.put("type", type);
+                averageAndThreshold(templateData, alert);
+                if (templateLocale != null) {
+                    plain = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_PLAIN + "." + type.name() + "."
+                            + templateLocale);
+                    html = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_HTML + "." + type.name() + "."
+                            + templateLocale);
+                } else {
+                    plain = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_PLAIN + "." + type.name());
+                    html = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_HTML + "." + type.name());
+                }
+            } else {
+                if (templateLocale != null) {
+                    plain = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_PLAIN + "." + templateLocale);
+                    html = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_HTML + "." + templateLocale);
+                } else {
+                    plain = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_PLAIN);
+                    html = getProp(props, defaultProps, EmailPlugin.PROP_TEMPLATE_HTML);
+                }
+            }
+
+            StringWriter writerPlain = new StringWriter();
+            StringWriter writerHtml = new StringWriter();
+            if (plain != null) {
+                ftlTemplate = Template.getPlainTextTemplate("plainTemplate", plain, ftlCfg);
+                ftlTemplate.process(templateData, writerPlain);
+            }  else {
+                ftlTemplatePlain.process(templateData, writerPlain);
+            }
+            if (html != null) {
+                ftlTemplate = Template.getPlainTextTemplate("htmlTemplate", html, ftlCfg);
+                ftlTemplate.process(templateData, writerHtml);
+            } else {
+                ftlTemplateHtml.process(templateData, writerHtml);
+            }
+
+            writerPlain.flush();
+            writerPlain.close();
+
+            writerHtml.flush();
+            writerHtml.close();
+
+            processed.put("plain", writerPlain.toString());
+            processed.put("html", writerHtml.toString());
+        }
+        return processed;
+    }
+
+    private void subjects(Map<String, Object> props, Alert alert) {
+        if (props != null && alert != null && alert.getTrigger() != null && getFirstCondition(alert) != null) {
+            Condition.Type type = getFirstConditionType(alert);
```

Comment:
Yes. This logic is designed for hawkular instead of a general alert.
Perhaps the best approach could be to have a plugin for hawkular specific scenarios and other plugin for general use cases.
I suggest to merge as it is, as our primary goal is hawkular integration, but within re-packaging tasks I'm going to create a more generic email plugin covering all possibilities (targetting more general use cases).
In any case, the logic is inside the plugin.